### PR TITLE
restore QtNDisplayButton

### PR DESCRIPTION
**Pull request by Czaki**
# Description

This PR restore QtNDisplayButton, mark it as deprecated, and exposes `QtStateButton` for future use

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/qt/__init__.py [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
 from .._qt.qt_viewer import QtViewer
-from .._qt.widgets.qt_viewer_buttons import QtViewerButtons
+from .._qt.widgets.qt_viewer_buttons import QtStateButton, QtViewerButtons
 from .threading import create_worker, thread_worker
 
+
+class QtNDisplayButton(QtStateButton):
+    def __init__(self, viewer):
+        warnings.warn(
+            "QtNDisplayButton is deprecated and will be removed in 0.4.9",
```
Comment (goanpeca) [inline napari/qt/__init__.py]: Hi @Czaki , thanks for working on this. Could you make the string localizable?

```python
from ..utils.translator import trans
...
trans._("QtNDisplayButton is deprecated and will be removed in 0.4.9")
```

Thanks!

**Review by goanpeca — CHANGES_REQUESTED** [review]
Small change request :)

File: napari/qt/__init__.py (hunk 1 above)
Comment (Czaki) [reply napari/qt/__init__.py]: Are you sure that warnings should be translated? I think that this should be only applied to the interface elements.

Comment (sofroniewn) [reply napari/qt/__init__.py]: It's probably helpful for people that they get translated, no? We might not have done it everywhere yet, but it seems reasonable to me. Thoughts @goanpeca?

**Review by sofroniewn — APPROVED** [review]
Sorry for missing this, I'll get the PR in and update the release notes before the next release.

I think adding the translation makes sense, but maybe @goanpeca can confirm. I also see no harm in it, so @Czaki if you can quickly do it then I'll merge.

**Comment by Czaki** [comment]
@sofroniewn @goanpeca I add a requested change. 
As napari documentation is whole in English then I think that such messages are for developers, not users, and should be in English.  But it could be discussed later and I do not want block this release. 

Call me when a new RC release was created. I manually will trigger test to check if everything works for me on CI.

**Comment by sofroniewn** [comment]
I just added a release note/ expanded deprecation note. I will merge when tests pass

**Pull request merged**